examples/client/Locomotion/src/pages/Profile/Code.js

```diff
@@ -58,6 +61,51 @@ const Code = () => {
       setLoading(false);
     }, [showErrorText]),
   );
+
+  useFocusEffect(
+    useCallback(() => {
+      setResendCounter(0);
+      setTimer(RESEND_SECONDS);
+      setIsCalling(false);
+    }, []),
+  );
+
+  const onLoginInternal = async (channel) => {
+    try {
+      await onLogin(user.phoneNumber, channel);
+    } catch (e) {
+      console.log('Bad login with response', e);
+      const status = e && e.response && e.response.status;
+      if (status === 429) {
+        setShowErrorText(i18n.t('login.tooManyRequestError'));
+      }
+    }
+  };
+
+  const onResendPress = async () => {
+    await onLoginInternal();
+    setResendCounter(currentValue => currentValue + 1);
+    setTimer(RESEND_SECONDS);
+  };
+
+  const onCallPress = async () => {
+    await onLoginInternal('call');
+
+    setIsCalling(true);
+  };
+
+  useEffect(() => {
+    const callingTimeout = setTimeout(() => {
+      setIsCalling(false);
+    }, 10 * 1000);
+
+    return () => {
+      if (callingTimeout) {
+        clearTimeout(callingTimeout);
+      }
+    };
+  }, [isCalling]);
```

Comment:
add check for isCalling value 
it would cause some render issues

Reply:
Tzodek